protocols/1f9218/capsule_filling.ot2.py

```diff
@@ -0,0 +1,90 @@
+from opentrons import labware, instruments, robot
+from otcustomizers import StringSelection
+import math
+
+metadata = {
+    'protocolName': 'Capsule Filling',
+    'author': 'Nick <protocols@opentrons.com>',
+    'source': 'Custom Protocol Request'
+}
+
+# create custom labware
+capsule_rack_name = 'milimoli_100_capsulerack'
+if capsule_rack_name not in labware.list():
+    labware.create(
+        capsule_rack_name,
+        grid=(10, 10),
+        spacing=(13, 13),
+        diameter=2,
+        depth=10,
+        volume=500
+    )
+
+# load modules and labware
+oil = labware.load('agilent_1_reservoir_290ml', '3', 'oil').wells(0)
+tips1000 = labware.load('opentrons_96_tiprack_1000ul', '6')
+
+
+def run_custom_protocol(
+        p1000_mount: StringSelection('right', 'left') = 'right',
+        number_of_capsules_to_fill: int = 100,
+        volume_to_fill_in_ul: float = 200
+):
+    # check
+    if volume_to_fill_in_ul < 100:
+        raise Exception('Invalid volume for P1000 pipette')
+
+    # load capsule plates
+    slots = ['4', '10'] if number_of_capsules_to_fill > 100 else ['4']
```

Comment:
Normally if the no tips are used in a specific tip rack, the tip rack will not show up on the app, so there's usually no need to check the number of tip racks to be loaded

Reply:
the slots calculation is actually for the number of capsules to fill, not for the tips (but thanks for the heads up about the tip racks regardless)